qp_klp/AmpliconStep.py

```diff
@@ -0,0 +1,102 @@
+from os import listdir, makedirs
+from os.path import join, isfile, basename
+from shutil import copyfile
+from qp_klp.Step import Step
+
+
+class AmpliconStep(Step):
+    def __init__(self, pipeline, master_qiita_job_id, sn_tid_map_by_project,
+                 status_update_callback=None):
+        super().__init__(pipeline,
+                         master_qiita_job_id,
+                         sn_tid_map_by_project,
+                         status_update_callback)
+
+        if pipeline.pipeline_type != 'amplicon':
+            raise ValueError("Cannot instantiate AmpliconStep object from "
+                             f"pipeline of type '{pipeline.pipeline_type}'")
+
+    def convert_bcl_to_fastq(self):
+        config = self.pipeline.configuration['bcl2fastq']
+        # note that pipeline.sample_sheet in this case
```

Comment:
This note makes sense but it would be better if it was clear that this is by design and historically (EMP primers)